web/components/navbar/index.js

```diff
@@ -23,43 +20,93 @@ const getButton = (buttonType) => {
   }
 };
 
-const Navbar = ({ displayType, buttonType }) => {
-  switch (displayType) {
-    case DISPLAY_TYPE.ONLY_LOGO:
-      return (
-        <nav>
-          <div>
-            <div>{getLogo()}</div>
-          </div>
-        </nav>
-      );
-    case DISPLAY_TYPE.LOGO_AND_BUTTON:
-      return (
-        <nav>
-          <div>
-            <div>{getLogo()}</div>
-          </div>
-          <div>
-            <div>{getButton(buttonType)}</div>
-          </div>
-        </nav>
-      );
-    default:
-      return (
-        <nav>
-          <div>
-            <div>{getLogo()}</div>
+class Navbar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hidden: false,
+      transparent: true,
+      lastY: 0,
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('scroll', this.handleScroll, { passive: true });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleScroll);
+  }
+
+  handleScroll = () => {
+    // Calculate scroll direction
+    const { lastY } = this.state;
+    const scrollingDown = (window.scrollY - lastY) >= 0;
+
+    // Calculate position
+    const offset = window.pageYOffset || document.documentElement.scrollTop;
+    const atTop = offset < 96;
```

Comment:
Can we store `96` into some kind of variable?

Reply:
Sure - 96 is in the CSS as navbar height, but I’ll replicate it here